app/code/Magento/MediaGallerySynchronization/Model/Filesystem/GetFileInfo.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronization\Model\Filesystem;
+
+use Magento\MediaGallerySynchronization\Model\Filesystem\FileInfoFactory;
+
+/**
+ * Get file information
+ */
+class GetFileInfo
+{
+    /**
+     * @var FileInfoFactory
+     */
+    private $fileInfoFactory;
+
+    /**
+     * GetFileInfo constructor.
+     * @param FileInfoFactory $fileInfoFactory
+     */
+    public function __construct(
+        FileInfoFactory $fileInfoFactory
+    ) {
+        $this->fileInfoFactory = $fileInfoFactory;
+    }
+
+    /**
+     * Get file information based on path provided.
+     *
+     * @param string $path
+     * @return FileInfo
+     */
+    public function execute(string $path): FileInfo
+    {
+        $splFileInfo = new \SplFileInfo($path);
```

Comment:
In the case of composition mentioned above, you will need to provide the only path to the factory and don't use SplFileInfo here.